### ISSUE TITLE
Invalid keys on OpenMandriva final release (Bugzilla Bug 344)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#344]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 344
Date: 2013-11-22 12:01:29 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;
To:   @berolinux
CC:   arisel@arisel.de, @benbullard79, @berolinux, @cris-b, gio_6b@yahoo.it, @itchka, @jclvanier, @tpgxyz

Depends on:   #580
Blocker for:  #284, #408, #514
Last updated: 2014-04-04 12:27:41 +0000

## Comment 2369
Date: 2013-11-22 12:01:29 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I know that there is the bug #284 regarding this question. I open a new bug because this is about the Final Release Candidate.

Version tested: Final Release Candidate, ISO build 2038, 64 bits.
Mode: Live, boot from USB.
Bug: I'm getting still packages with invalid keys ID:

**********
# urpmi vidalia
Para satisfacer las dependencias, se instalarán los paquetes siguientes:     
 Paquete                        Versión      Revisión      Dist  DEpoch Arq. 
(soporte «contrib (Oxygen2013.0-3)»)                                         
 lib64tsocks1                   1.8          0.beta5.7     omv   2013.0 x86_64 
 tor                            0.2.2.39     2             omv   2013.0 x86_64  
 tsocks                         1.8          0.beta5.7     omv   2013.0 x86_64  
 vidalia                        0.2.15       1             omv   2013.0 x86_64  
se utilizarán 26MB de espacio de disco adicional.
se descargarán 4.4MB de paquetes.
¿Efectuar la instalación de los 4 paquetes? (S/n) s   
$MIRRORLIST: media/contrib/release/tor-0.2.2.39-2-omv2013.0.x86_64.rpm
$MIRRORLIST: media/contrib/release/lib64tsocks1-1.8-0.beta5.7-omv2013.0.x86_64.rpm
$MIRRORLIST: media/contrib/release/vidalia-0.2.15-1-omv2013.0.x86_64.rpm
$MIRRORLIST: media/contrib/release/tsocks-1.8-0.beta5.7-omv2013.0.x86_64.rpm
                                                                                Los siguientes paquetes contienen firmas no válidas:
/var/cache/urpmi/rpms/lib64tsocks1-1.8-0.beta5.7-omv2013.0.x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 23:06:03 ART, Key ID fc641c607bf07c6d))
/var/cache/urpmi/rpms/tsocks-1.8-0.beta5.7-omv2013.0.x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 23:06:03 ART, Key ID fc641c607bf07c6d))
/var/cache/urpmi/rpms/vidalia-0.2.15-1-omv2013.0.x86_64.rpm: ID de clave no válida (OK (DSA/SHA1, sáb 20 abr 2013 02:44:14 ART, Key ID 07557a5c70e7dfbd))
¿Desea continuar con la instalación? (s/N) s
**********

Sorry for not add "LC_ALL=C", but I just didn't expect this kind of problem. In any case "ID de clave no válida" means "Invalid key ID".

## Comment 2371
Date: 2013-11-22 12:07:22 +0000
From: arisel &lt;<arisel@arisel.de>&gt;

by taking a look: tsocks has not been rebuild for 2013.0. 
I'd suggest a simple rebuild so that it gets the right keys.

Anyone from QA?

## Comment 2816
Date: 2013-12-23 05:50:04 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

The four files needed to solve the bug #403 gave the "Medium without key" warning. The problem is like bug #284, but in the Final Release.

The files are:
- x11-driver-video-nvidia-current-331.20-6-omv2013.0.x86_64.rpm
- dkms-nvidia-current-331.20-6-omv2013.0.x86_64.rpm
- nvidia-current-doc-html-331.20-6-omv2013.0.x86_64.rpm
- x11-driver-video-nvidia-current-32bit-331.20-6-omv2013.0.x86_64.rpm

## Comment 2824
Date: 2013-12-23 14:30:23 +0000
From: @tpgxyz

(In reply to comment #1)
> by taking a look: tsocks has not been rebuild for 2013.0. 
> I'd suggest a simple rebuild so that it gets the right keys.
> 

tsocks got rebuild and pushed to mirrors.

> Anyone from QA?

No :p

## Comment 3004
Date: 2014-01-22 23:29:01 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Another package with invalid key:

megamario-1.7-1-omv2013.0.x86_64

## Comment 3005
Date: 2014-01-23 00:31:22 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Another one:

extra-cmake-modules-0.0.1-1-omv2013.0.noarch

## Comment 3008
Date: 2014-01-23 09:50:15 +0000
From: @tpgxyz

We are currently testing a mechanism which will allow us to resign all packages.

## Comment 3069
Date: 2014-02-01 14:44:49 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I list here more packages to check about invalid keys or packages unsigned. I know that it's a work in progress. I just add the list to verify that all the packages will end up as signed and with valid keys.

Packages with invalid keys:
- drakbackup-0.19.3-2-omv2013.0.noarch.rpm
- drakvirt-0.8.2-3-omv2013.0.noarch.rpm

## Comment 3131
Date: 2014-02-06 13:29:33 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

These packages have invalid keys, but the novelty is that they are from the just-appeared update. The files are dated on February 6th on the repository (http://abf-downloads.rosalinux.ru/openmandriva2013.0/repository/x86_64/media/main/updates/), so the problem still exists in "brand new" packages:

- avahi-0.6.31-14-omv2013.0.x86_64.rpm
- lib64avahi-client3-0.6.31-14-omv2013.0.x86_64.rpm
- lib64avahi-common3-0.6.31-14-omv2013.0.x86_64.rpm
- lib64avahi-core7-0.6.31-14-omv2013.0.x86_64.rpm
- lib64avahi-glib1-0.6.31-14-omv2013.0.x86_64.rpm

## Comment 3136
Date: 2014-02-06 21:07:42 +0000
From: @itchka

What is even worse is that this package should not be in the main repository yet it is in testing and has not been released and would not have been if it had an invalid signature.

Colin Close
QA Team

## Comment 3138
Date: 2014-02-06 21:54:21 +0000
From: @itchka

What is even worse is that this package should not be in the main repository yet it is in testing and has not been released and would not have been if it had an invalid signature.
However between the time of me writing most of this reply and the notification of the ABF  maintainer the signing bug has been fixed. How's that for service!!!

Colin Close
QA Team

## Comment 3139
Date: 2014-02-06 22:26:50 +0000
From: @jclvanier

Suggested name for our next distro: "shooting star" the woman who shoots faster than his shadow.

## Comment 3140
Date: 2014-02-07 07:54:06 +0000
From: @tpgxyz

(In reply to comment #8)
> These packages have invalid keys, but the novelty is that they are from the
> just-appeared update. 

Looks like there were some timeouts on our new servers. avokhmin fixed this, so what is left is to resign this package.

## Comment 3253
Date: 2014-02-18 10:34:46 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Still no key for get-skype. Is this an "external package"?

/get-skype-4.2.0.11-6-omv2013.0.noarch.rpm: Medium without key (OK (RSA/SHA1, Sun 17 Nov 2013 15:02:21 CET, Key ID 180922d8bf81de15))

## Comment 3257
Date: 2014-02-18 13:22:07 +0000
From: @tpgxyz

(In reply to comment #13)
> Still no key for get-skype. Is this an "external package"?
> 
> /get-skype-4.2.0.11-6-omv2013.0.noarch.rpm: Medium without key (OK
> (RSA/SHA1, Sun 17 Nov 2013 15:02:21 CET, Key ID 180922d8bf81de15))

I've just regenerated repo, pleas try again in few hours.

## Comment 3277
Date: 2014-02-22 15:10:14 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Another package with invalid key:

- wifi-radar-1.9.9-2-omv2013.0.noarch.rpm

## Comment 3284
Date: 2014-02-24 21:56:27 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

Bug #514 also refers packages with invalid keys.

## Comment 3285
Date: 2014-02-24 21:57:35 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

(In reply to comment #14)
> > /get-skype-4.2.0.11-6-omv2013.0.noarch.rpm: Medium without key (OK
> > (RSA/SHA1, Sun 17 Nov 2013 15:02:21 CET, Key ID 180922d8bf81de15))
> 
> I've just regenerated repo, pleas try again in few hours.

Still no key.

## Comment 3292
Date: 2014-02-25 08:51:45 +0000
From: @tpgxyz

*** Bug #514 has been marked as a duplicate of this bug. ***

## Comment 3293
Date: 2014-02-25 08:56:53 +0000
From: @tpgxyz

Current situation:

www.abf.io have fixed rpm signing feature, so during publishing packages are signed with OMV key. Also while regenerating repository ABF resigns all rpm files which are missing gpg key. 
Looks like there is some bogus, which is hard to investigate.

Bero is it possible for you to log into ABF and force resign all files inside repository ?

## Comment 3304
Date: 2014-02-25 09:24:33 +0000
From: @tpgxyz

*** Bug #408 has been marked as a duplicate of this bug. ***

## Comment 3306
Date: 2014-02-25 09:25:22 +0000
From: @tpgxyz

*** Bug #284 has been marked as a duplicate of this bug. ***

## Comment 3345
Date: 2014-02-26 13:48:31 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I add more packages to my system today (Feb 26). There are packages with the "Medium without key" problem:

- faac-1.28-6-plf2013.0.x86_64
- faad2-2.7-5-omv2013.0.x86_64
- lib64dvdcss2-1.2.13-1-plf2013.0.x86_64
- lib64faac0-1.28-6-plf2013.0.x86_64
- lib64faad2_2-2.7-5-omv2013.0.x86_64

Some of them belongs to the "Restricted" repository, I think that all the repo may have the problem. I'm not so sure in the faad2/lib64faad2_2 case.

I don't fill this as a new bug, as in a previous comment (Comment 2) I already said about missing (not invalid) keys.

## Comment 3346
Date: 2014-02-26 13:53:48 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

I forgot these two:

- lib64x264_133-0.133-0.20130731.2-omv2013.0.x86_64
- x264-0.133-0.20130731.2-omv2013.0.x86_64

Also "medium without key".

## Comment 3353
Date: 2014-02-26 21:40:00 +0000
From: @tpgxyz

ABF guys run rpm --resign on 2013.0 repo today.

Please make sure your mirror is updated, and try to check these packages if they are signed.

## Comment 3623
Date: 2014-03-05 21:41:00 +0000
From: @tpgxyz

(In reply to comment #23)
> I forgot these two:
> 
> - lib64x264_133-0.133-0.20130731.2-omv2013.0.x86_64
> - x264-0.133-0.20130731.2-omv2013.0.x86_64
> 
> Also "medium without key".

Downloaded form our official mirrors.

[root@tpgbox sources]# rpm -K x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm 
x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm: rsa sha1 md5 OK

Seems to be fixed right ?

## Comment 3636
Date: 2014-03-06 07:24:22 +0000
From: @itchka

root@localhost rpms]# urpmi x264
Packages lib64avcodec54-1.2.1-1-omv2013.0.x86_64, glibc-2.19-1-omv2013.0.x86_64, lib64avutil52-1.2.1-1-omv2013.0.x86_64, lib64x11_6-1.6.2-1-omv2013.0.x86_64, lib64avformat54-1.2.1-1-omv2013.0.x86_64, lib64swscaler2-1.2.1-1-omv2013.0.x86_64 are already installed

    $MIRRORLIST: media/restricted/release/x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm
The following package has bad signature:                                       
/var/cache/urpmi/rpms/x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm: Medium without key (OK (RSA/SHA1, Wed 26 Feb 2014 13:24:53 GMT, Key ID 180922d8bf81de15))
Do you want to continue installation ? (y/N) 

Not fixed as far as I can see..

P.S. I made sure that any previous versions were removed from the cache.

## Comment 3641
Date: 2014-03-06 10:47:36 +0000
From: @tpgxyz

(In reply to comment #26)
> 
> 

Please follow below routine:
1. download x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm from yandex.ru

wget ftp://mirror.yandex.ru/openmandriva/openmandriva2013.0/repository/x86_64/restricted/release/x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm

2. Run rpm key check on downloaded file
rpm -K x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm

> /var/cache/urpmi/rpms/x264-0.133-0.20130731.2-omv2013.0.x86_64.rpm: Medium
> without key (OK (RSA/SHA1, Wed 26 Feb 2014 13:24:53 GMT, Key ID
> 180922d8bf81de15))
> Do you want to continue installation ? (y/N) 
> 
> Not fixed as far as I can see..

Remove all old urpmi configuration from /var/cache/urpmi, /var/lib/urpmi and /etc/urpmi.d

then run urpmi.removemedia -a and then urpmi.addmedia --distrib --mirrorlist '$MIRRORLIST'

## Comment 4023
Date: 2014-03-16 03:59:57 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

The Neverending Story:

Update of March 16th, 3:45 UT.

The following warning appears (approx. translation from Spanish to English):

----------
The following packages have invalid signatures:

dkms-nvidia-current-334.21-1-omv2013.0.x86_64.rpm: Medium without key [...]
nvidia-current-cuda-opencl-334.21-1-omv2013.0.x86_64.rpm: Medium without key [...]
nvidia-current-doc-html-334.21-1-omv2013.0.x86_64.rpm: Medium without key [...]
x11-driver-video-nvidia-current-32bit-334.21-1-omv2013.0.x86_64.rpm: Medium without key [...]
x11-driver-video-nvidia-current-334.21-1-omv2013.0.x86_64.rpm: Medium without key [...]

Do you want to continue installation ? (y/N) y
----------

## Comment 4337
Date: 2014-03-24 15:08:13 +0000
From: Gus Ballan &lt;<siriustheking@yahoo.com>&gt;

This problem still exists in the OpenMandriva 2014 Beta. See bug #606.

## Comment 4458
Date: 2014-03-25 16:28:01 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Confirmed in beta1 64 bit.

## Comment 4736
Date: 2014-04-01 13:48:15 +0000
From: Giorgio &lt;<gio_6b@yahoo.it>&gt;

Still valid.
Updated today more than 80 packets in OMV 2014 beta1 64 bit. Almost all are "Medium without key".

## Comment 4737
Date: 2014-04-01 14:19:24 +0000
From: @tpgxyz

(In reply to comment #31)
> Still valid.
> Updated today more than 80 packets in OMV 2014 beta1 64 bit. Almost all are
> "Medium without key".

It is not packages it is because medium is not signed, which is false statement, because perl-URPM have troubles when importing pubkey from mirrors.
See bug https://issues.openmandriva.org/show_bug.cgi?id=627

## Comment 4798
Date: 2014-04-03 05:40:50 +0000
From: @cris-b

fixed in perl-URPM 4.48.6

## Comment 4859
Date: 2014-04-04 02:23:54 +0000
From: @benbullard79

Hmmm. I must be reading something wrong...

found package(s): joe-3.7-9-omv2013.0.x86_64 joe-3.7-10-omv2014.0.x86_64
Packages glibc-2.19-2-omv2014.0.x86_64, lib64ncursesw5-5.9-7.20130218.4-omv2014.0.x86_64 are already installed

retrieving rpm files from medium "main (Phosphorus2014.0-1)"...
    $MIRRORLIST: media/main/release/joe-3.7-10-omv2014.0.x86_64.rpm
retrieved $MIRRORLIST media/main/release joe-3.7-10-omv2014.0.x86_64.rpm                                                                                                                     
...retrieving done
The following package has bad signature:
/var/cache/urpmi/rpms/joe-3.7-10-omv2014.0.x86_64.rpm: Medium without key (OK (RSA/SHA1, Tue 11 Feb 2014 07:16:44 AM CST, Key ID 180922d8bf81de15))
Do you want to continue installation ? (y/N)

# rpm -qa perl-URPM
perl-URPM-4.48.7-1-omv2014.0.x86_64

## Comment 4860
Date: 2014-04-04 02:25:14 +0000
From: @benbullard79

Hmmm. I must be reading something wrong...

found package(s): joe-3.7-9-omv2013.0.x86_64 joe-3.7-10-omv2014.0.x86_64
Packages glibc-2.19-2-omv2014.0.x86_64, lib64ncursesw5-5.9-7.20130218.4-omv2014.0.x86_64 are already installed

retrieving rpm files from medium "main (Phosphorus2014.0-1)"...
    $MIRRORLIST: media/main/release/joe-3.7-10-omv2014.0.x86_64.rpm
retrieved $MIRRORLIST media/main/release joe-3.7-10-omv2014.0.x86_64.rpm                                                                                                                     
...retrieving done
The following package has bad signature:
/var/cache/urpmi/rpms/joe-3.7-10-omv2014.0.x86_64.rpm: Medium without key (OK (RSA/SHA1, Tue 11 Feb 2014 07:16:44 AM CST, Key ID 180922d8bf81de15))
Do you want to continue installation ? (y/N)

# rpm -qa perl-URPM
perl-URPM-4.48.7-1-omv2014.0.x86_64

## Comment 4861
Date: 2014-04-04 02:25:44 +0000
From: @benbullard79

Confirming

## Comment 4863
Date: 2014-04-04 06:37:08 +0000
From: @tpgxyz

(In reply to comment #36)
> Confirming

Rebuilding joe.

## Comment 4882
Date: 2014-04-04 12:27:41 +0000
From: @benbullard79

Todays update went through with no errors so this does appear to be fixed.

Many thanks.
